TGC/decisions/GRE2.txt

```diff
@@ -0,0 +1,693 @@
+			 all_core = { owned_by = THIS }			  
+			  OR = {
+					has_country_flag = absolute_monarchy_gov
+					has_country_flag = semi_constitutional_monarchy_gov
+					has_country_flag = constitutional_monarchy_gov
+					OR = { government = fascist_dictatorship }
+				  }
+		  }
+		  effect = {
+			  prestige = 10.5
+			  badboy = 22
+			  set_country_flag = byzantine_levant
+			  EGY_897 = { add_core = BYZ }
+			  EGY_903 = { add_core = BYZ }
+			  EGY_915 = { add_core = BYZ }
+			  EGY_921 = { add_core = BYZ }
+			  909 = { add_core = BYZ }
+			  908 = { add_core = BYZ }
+			  }
+			  
+		 ai_will_do {
+			 factor = 1
+			 modifier = {
+				 factor = 0
+				 badboy = 10
+			 }
+		 }
+	}
+	#Description: Accept Assyrians culture if you own Assyria proper.
+	 accept_assyrians = {
+		 potential = {
+			 primary_culture = greek
+			 OR = {
+				 tag = BYZ
+			     tag = GRE
+			  }
+			  NOT = { has_country_flag = accepted_assyrians }
+		  }
+
+		  allow = {
+			  war = no
+			 is_disarmed = no
+			OR = {
+				 OR = { has_country_flag = absolute_monarchy_gov }
+				 OR = { has_country_flag = semi_constitutional_monarchy_gov }
+				 OR = { has_country_flag = constitutional_monarchy_gov }
+				 OR = { has_country_flag = democracy_gov }
+			 }
+			 all_core = { owned_by = THIS }
+			 926 = { owned_by = THIS }
+			 TUR_923 = { owned_by = THIS }
+			 nationalism_n_imperialism = 1
+		 }
+
+		 effect = {
+		     badboy = -5
+			 set_country_flag = accepted_assyrians
+			 add_accepted_culture = assyrian 
+			 any_pop = {
+				 limit = { has_pop_culture = assyrian }
+				 militancy = -3
+		    }
+		  }
+
+		 ai_will_do ={ factor = 1 }
+	 }
+	 
+	 #Description: Accept Coptic culture if you own Egypt proper.
+	 accept_copts = {
+		 potential = {
+			 primary_culture = greek
+			 OR = {
+				 tag = BYZ
+			     tag = GRE
+			  }
+			  NOT = { has_country_flag = accepted_copts }
+		  }
+
+		  allow = {
+			  war = no
+			 is_disarmed = no
+			OR = {
+				 OR = { has_country_flag = absolute_monarchy_gov }
+				 OR = { has_country_flag = semi_constitutional_monarchy_gov }
+				 OR = { has_country_flag = constitutional_monarchy_gov }
+				 OR = { has_country_flag = democracy_gov }
+			 }
+			 all_core = { owned_by = THIS }
+			 EGY_1751 = { owned_by = THIS }
+			 EGY_1745 = { owned_by = THIS }
+			 EGY_1762 = { owned_by = THIS }
+			 EGY_1771 = { owned_by = THIS }
+			 nationalism_n_imperialism = 1
+		 }
+
+		 effect = {
+			 set_country_flag = accepted_copts
+			 add_accepted_culture = copt 
+			 badboy = -5
+			 any_pop = {
+				 limit = { has_pop_culture = copt }
+				 militancy = -3
+		    }
+		  }
+
+		 ai_will_do ={ factor = 1 }
+	 }
+	 #Description: Byzantine ambitions towards Messopotamia.
+		reclaim_messopotamia = {
+		potential = {
+			 primary_culture = greek
+			 tag = BYZ
+			 NOT = { has_country_flag = retreated_from_anatolia }
+			  has_country_flag = legacy_of_rome
+			  has_country_flag = no_turks_in_anatolia
```

Comment:
The country_flag byzantine_levant needs the country_flag legacy_of_rome to exist, there's no point in checking for legacy_of_rome here again, it was already checked at the start of the event chain unless you can somehow lose this flag as a possible outcome of the event. Most event chains can be controlled by one flag once it gets rolling.

There's other instances of this in the code and with other flags, once again the fewer checks we make the better it is for performance

Reply:
With my latest comit it should now be fixed

Reply:
My nit picks are fixed indeed.